### PR TITLE
Fix(optimizer): traverse UNNEST scope

### PR DESCRIPTION
**Pull request by GeorgeSittas**

### CODE REVIEW

File: sqlglot/optimizer/scope.py [hunk 1]
```diff
@@ -711,6 +711,31 @@ def _traverse_subqueries(scope):
         scope.subquery_scopes.append(top)
 
 
+def _traverse_udtfs(scope):
+    sources = {}
+
+    if isinstance(scope.expression, exp.Unnest):
```
Comment (tobymao) [inline sqlglot/optimizer/scope.py:717]: why are you checking specifically for unnest?

File: sqlglot/optimizer/scope.py [hunk 2]
```diff
@@ -711,6 +711,31 @@ def _traverse_subqueries(scope):
...
+        unnested_source = scope.expression.expressions[0]
```
Comment (tobymao) [inline sqlglot/optimizer/scope.py:718]: why only the first value?

File: sqlglot/optimizer/scope.py (hunk 1 above)
Comment (GeorgeSittas) [reply sqlglot/optimizer/scope.py:717]: I'll work on handling `exp.Lateral` too. I think these two are the only meaningful `UDTF`s to deal with here, as the other one is `Values`.

File: sqlglot/optimizer/scope.py (hunk 2 above)
Comment (GeorgeSittas) [reply sqlglot/optimizer/scope.py:718]: Will update to traverse all expressions

File: sqlglot/parser.py [hunk 3]
```diff
@@ -2242,7 +2242,12 @@ def _parse_lateral(self) -> t.Optional[exp.Lateral]:
             return None
 
         if not this:
-            this = self._parse_function() or self._parse_id_var(any_token=False)
+            this = (
+                self._parse_unnest()
```
Comment (GeorgeSittas) [inline sqlglot/parser.py:2246]: Without this we used to parse the `UNNEST` in `... JOIN LATERAL UNNEST(...)` into `exp.Anonymous`.

**Pull request merged**